frontend/src/layout/navigation/TopBar/notificationsLogic.ts

```diff
@@ -0,0 +1,94 @@
+import { actions, events, kea, listeners, path, reducers, selectors } from 'kea'
+import { loaders } from 'kea-loaders'
+import api from 'lib/api'
+import { teamLogic } from 'scenes/teamLogic'
+import { ActivityLogItem, humanize, HumanizedActivityLogItem } from 'lib/components/ActivityLog/humanizeActivity'
+
+import type { notificationsLogicType } from './notificationsLogicType'
+import { describerFor } from 'lib/components/ActivityLog/activityLogLogic'
+
+const POLL_TIMEOUT = 5000
+const MARK_READ_TIMEOUT = 15000
+
+export const notificationsLogic = kea<notificationsLogicType>([
+    path(['layout', 'navigation', 'TopBar', 'notificationsLogic']),
+    actions({
+        toggleNotificationsPopover: true,
+        togglePolling: (pageIsVisible: boolean) => ({ pageIsVisible }),
+        setPollTimeout: (pollTimeout: number) => ({ pollTimeout }),
+        setMarkReadTimeout: (markReadTimeout: number) => ({ markReadTimeout }),
+    }),
+    loaders(({ actions, values }) => ({
+        importantChanges: [
+            [] as HumanizedActivityLogItem[],
+            {
+                loadImportantChanges: async (_, breakpoint) => {
+                    await breakpoint(1)
+
+                    clearTimeout(values.pollTimeout)
+
+                    const response = (await api.get(
```

Comment:
If the network request fails can this throw? If so do we rather want to wrap it in a try/finally and have the timeout recreation in the finally block?